src/imu_diag_ros_publisher.cpp

```diff
@@ -59,17 +59,26 @@ void ImuDiagRosPublisher::run()
        // std::cout << "thread " << this_id << " running...\n";
        // RCLCPP_INFO(rclcpp::get_logger("rclcpp"), "running: '%d'", this_id);
 
-        //auto started = std::chrono::high_resolution_clock::now();
-        m_message = m_dataProvider->getData(count);
-        //auto done = std::chrono::high_resolution_clock::now();
+        int32_t operation_mode = m_node->get_parameter("operation_mode").get_parameter_value().get<int32_t>();
 
-        RCLCPP_INFO(rclcpp::get_logger("rclcpp_imu_diag_data"), "Publishing imu diag data: lost_samples_count = '%d' diag_checksum_error_flag= '%d' flash_counter = '%d' ",
-                    m_message.lost_samples_count, m_message.diag_checksum_error_flag, m_message.flash_counter);
+        if(operation_mode == SAMPLING_MODE)
+        {
+            //auto started = std::chrono::high_resolution_clock::now();
+            m_message = m_dataProvider->getData(count);
+            //auto done = std::chrono::high_resolution_clock::now();
 
-        m_publisher->publish(m_message);
-        count++;
-        //rclcpp::spin_some(m_node);
-        loopRate.sleep();
+            RCLCPP_INFO(rclcpp::get_logger("rclcpp_imu_diag_data"), "Publishing imu diag data: lost_samples_count = '%d' diag_checksum_error_flag= '%d' flash_counter = '%d' ",
+                        m_message.lost_samples_count, m_message.diag_checksum_error_flag, m_message.flash_counter);
+
+            m_publisher->publish(m_message);
+            count++;
+            loopRate.sleep();
+        }
+        else
+            if(operation_mode == STOP_LOOP)
+                break;
```

Comment:
use switch case for operation_mode values handling, please perform this change everywhere in the code where different operations are done based on the value of operation_mode